examples/dynamic-client/cluster_scoped_custom_resource.py

```diff
@@ -0,0 +1,152 @@
+    crd_api = client.resources.get(
+        api_version="apiextensions.k8s.io/v1beta1", kind="CustomResourceDefinition"
+    )
+
+    # Creating a Namespaced CRD named "ingressroutes.apps.example.com"
+    name = "ingressroutes.apps.example.com"
+
+    crd_manifest = {
+        "apiVersion": "apiextensions.k8s.io/v1beta1",
+        "kind": "CustomResourceDefinition",
+        "metadata": {
+            "name": name,
+        },
+        "spec": {
+            "group": "apps.example.com",
+            "names": {
+                "kind": "IngressRoute",
+                "listKind": "IngressRouteList",
+                "plural": "ingressroutes",
+                "singular": "ingressroute",
+            },
+            "scope": "Cluster",
+            "version": "v1",
+            "subresources": {"status": {}},
+        },
+    }
+
+    crd_creation_respone = crd_api.create(crd_manifest)
+    print(
+        "\n[INFO] custom resource definition `ingressroutes.apps.example.com` created\n"
+    )
+    print("%s\t\t%s" % ("SCOPE", "NAME"))
+    print(
+        "%s\t\t%s\n"
+        % (crd_creation_respone.spec.scope, crd_creation_respone.metadata.name)
+    )
+
+    # Fetching the "ingressroutes" CRD api
+
+    try:
+        ingressroute_api = client.resources.get(
+            api_version="apps.example.com/v1", kind="IngressRoute"
+        )
+    except ResourceNotFoundError:
+        # Need to wait a sec for the discovery layer to get updated
+        time.sleep(2)
+
+    ingressroute_api = client.resources.get(
+        api_version="apps.example.com/v1", kind="IngressRoute"
+    )
+
+    # Creating a custom resource (CR) `ingress-route-*`, using the above CRD `ingressroutes.apps.example.com`
+
+    ingressroute_manifest_one = {
+        "apiVersion": "apps.example.com/v1",
+        "kind": "IngressRoute",
+        "metadata": {
+            "name": "ingress-route-one",
+        },
+        "spec": {},
+    }
+
+    ingressroute_manifest_second = {
+        "apiVersion": "apps.example.com/v1",
+        "kind": "IngressRoute",
+        "metadata": {
+            "name": "ingress-route-second",
+        },
+        "spec": {},
+    }
+
+    ingressroute_api.create(body=ingressroute_manifest_one)
+    ingressroute_api.create(body=ingressroute_manifest_second)
+    print("\n[INFO] custom resources `ingress-route-*` created\n")
+
+    # Listing the `ingress-route-*` custom resources
+
+    ingress_routes_list = ingressroute_api.get()
+    print("%s\t\t\t\t%s" % ("NAME", "SPEC"))
+    for item in ingress_routes_list.items:
+        print("%s\t\t%s" % (item.metadata.name, item.spec))
+
+    # Patching the ingressroutes custom resources
+
+    ingressroute_manifest_one["spec"]["entrypoints"] = ["websecure"]
+    ingressroute_manifest_second["spec"]["entrypoints"] = ["web"]
+
+    patch_ingressroute_one = ingressroute_api.patch(
+        body=ingressroute_manifest_one, content_type="application/merge-patch+json"
```

Comment:
better to use strategic-merge-patch

Reply:
Hi @yliaog, thanks for all your useful pointers.
In case of applying a patch on a custom_object with `content_type="application/strategic-merge-patch"` as you suggested, 

```
   patch_ingressroute_second = ingressroute_api.patch(
        body=ingressroute_manifest_second, content_type="application/strategic-merge-patch+json"
    )
```
it throws the following error:

```
Reason: Unsupported Media Type
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'a7b2fb82-0b68-40ee-8439-183cc9c1c0f0', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': 'd4c6d349-a9e2-454c-b88f-5beb06b64c0b', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'fa2d2d6c-312f-41bb-bc32-d00c53695884', 'Date': 'Mon, 03 May 2021 07:14:43 GMT', 'Content-Length': '293'})
HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json, application/apply-patch+yaml","reason":"UnsupportedMediaType","code":415}\n'
```
I also see in the `CustomObjectApi` not mentioning `application/strategic-merge-patch` anywhere in the `content-type` headers: https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CustomObjectsApi.md#http-request-headers-18

Could you please point me if `content_type="application/strategic-merge-patch"` is actually not supported & needs to be implemented in the `CustomObjectApi` or I'm doing something wrong here?

Reply:
https://github.com/kubernetes-client/python/commit/789e37464b8512ee4687ecf03221ff088afde450#diff-5a785a7fc5b19b934975d4d49eb71d730de127362b35f89e368179b8355796c8 is the patch to support strategic merge patch.

@roycaihw do you know why is this error?

Reply:
@yliaog @roycaihw just adding information

The `kubernetes` python module (I'm using locally) is version `12.0.1`

```
>>> import kubernetes
>>> kubernetes.__version__
'12.0.1'
```

Reply:
ok, strategic merge patch is not supported currently for custom resources

Reply:
Thank you @yliaog, for confirming.

One question: as you pointed in the comments above, that there is already a patch for supporting `strategic-merge-patch`. Is that something that would be available in the upcoming releases? Or that is not a patch for supporting `strategic-merge-patch` using `dynamic-client`?

Thank you.

Reply:
understood. Thank you. :)